magma/debug.py

```diff
@@ -18,7 +20,20 @@ class _DebugInfo:
 config.register(use_uinspect=RuntimeConfig(True))
 
 
+_extra_frames_to_skip_stack = Stack()
+
+
+def _get_extra_frames_to_skip_stack():
+    global _extra_frames_to_skip_stack
+    return _extra_frames_to_skip_stack
+
+
 def get_debug_info(frames_to_skip):
+    try:
+        extra_frames_to_skip = _get_extra_frames_to_skip_stack().peek()
+    except IndexError:
+        extra_frames_to_skip = 0
```

Comment:
Could be useful to add a `safe_peek` method to the `Stack` which takes a default argument
```suggestion
    extra_frames_to_skip = _get_extra_frames_to_skip_stack().safe_peek(default=0)
```